test/src/main/java/com/coderanch/test/AbstractObjectTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.test;
+
+import org.junit.Rule;
+import org.junit.experimental.theories.FromDataPoints;
+import org.junit.experimental.theories.Theory;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeNotNull;
+import static org.junit.Assume.assumeThat;
+
+/**
+ * Tests the common methods of all objects.
+ */
+@SuppressWarnings("squid:S00100")
+public abstract class AbstractObjectTest {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Convert this "AbstractObjectTest" class to a concrete class with a private constructor ([squid:S1694](https://sonarcloud.io/organizations/coderanch-corral/rules#rule_key=squid%3AS1694))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.coderanch%3Ablackjack-parent&pullRequest=11&issues=AWaCb4Gtn3WdA7VnJdq1&open=AWaCb4Gtn3WdA7VnJdq1)

[ISSUE_KEY]: AWaCb4Gtn3WdA7VnJdq1